community/server/src/main/java/org/neo4j/server/rrd/RrdFactory.java

```diff
@@ -71,28 +69,37 @@ public RrdFactory( Config config, Logging logging )
         this.log = logging.getConsoleLog( getClass() );
     }
 
-    public org.neo4j.server.database.RrdDbWrapper createRrdDbAndSampler( final Database db, JobScheduler scheduler ) throws IOException
+    public org.neo4j.server.database.RrdDbWrapper createRrdDbAndSampler( final Database db, JobScheduler scheduler )
+            throws IOException
     {
         NeoStoreProvider neoStore = db.getGraph().getDependencyResolver().resolveDependency( NeoStoreProvider.class );
+        SampleableDatabaseGuard sampleableDatabaseGuard = new SampleableDatabaseGuard();
```

Comment:
why do we need to introduce `SampleableDatabaseGuard` and not just use already available `AvailabilityGuard`. It looks like we only availability flag?